datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/util/LabelUtils.java

```diff
@@ -553,10 +559,20 @@ private static GlossaryTermAssociationArray removeTermsIfExists(GlossaryTerms te
     return termAssociationArray;
   }
 
-  private static void ingestChangeProposals(List<MetadataChangeProposal> changes, EntityService entityService, Urn actor) {
+  private static void ingestChangeProposals(List<MetadataChangeProposal> changes, EntityService entityService, Urn actor, String eTag) {
     // TODO: Replace this with a batch ingest proposals endpoint.
+    Map<String, Long> createdOnMap = ETagUtil.extractETag(eTag);
     for (MetadataChangeProposal change : changes) {
-      entityService.ingestProposal(change, getAuditStamp(actor), false);
+      if (createdOnMap.containsKey(change.getEntityUrn())) {
+        entityService.ingestProposal(change, getAuditStamp(actor), false, createdOnMap.get(change.getEntityUrn()));
+      } else {
+        entityService.ingestProposal(change, getAuditStamp(actor), false);
+      }
+
+      // ETAG Comment: If - Esle branch from about should be deleted and only if branch kept.
```

Comment:
Can we mark the eTag as Nullable as we pass it around if this is the intent? So we know that it might be null, and will be for all current use cases, as it goes down the stack.

Reply:
Yes